### ISSUE TITLE
💅 Biome replaces isNan by Number.isNan, which is a different function

### ISSUE THREAD

**Issue by yann-combarnous** [biomejs/biome#2075]
### Environment information

```bash
When applying fixes, I noticed Biome replaced isNan by Number.isNan, but according to 
https://www.w3schools.com/jsref/jsref_isnan_number.asp, these are not equivalent.
```

### Rule name

useNumberNamespace

### Playground link

n/a

### Expected result

Do not use Number namespace for isNan function.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
@yann-combarnous please provide a playground link

**Comment by yann-combarnous** [comment]
https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABpAHMAVgBhAGwAaQBkAFAAcgBpAGMAZQAoAHYAYQBsAHUAZQA6ACAAdQBuAGsAbgBvAHcAbgApADoAIABUAFYAYQBsAGkAZABhAHQAbwByAFIAZQBzAHUAbAB0ACAAewAKACAAIAByAGUAdAB1AHIAbgAgAGkAcwBOAGEATgAoAHYAYQBsAHUAZQApAAoAIAAgACAAIAA%2FACAAaQAxADgAbgAuAHQAKAAnAHYAYQBsAGkAZABhAHQAbwByAC4AaQBuAHYAYQBsAGkAZABOAHUAbQBiAGUAcgAnACkACgAgACAAIAAgADoAIAB1AG4AZABlAGYAaQBuAGUAZAA7AAoAfQA%3D

**Comment by arendjr** [comment]
Indeed the playground shows the diagnostic is coming from the [lint/suspicious/noGlobalIsNan](https://biomejs.dev/linter/rules/no-global-is-nan) rule. The documentation also seems to clearly highlight that the two functions are not the same, and the fix is properly marked as unsafe, meaning it should be manually reviewed before applying.

As far as I can see, this is all working and documented as intended. @yann-combarnous If this is not what you want, I think the suggested action is to disable the `noGlobalIsNan` rule.

**Comment by yann-combarnous** [comment]
Fair enough, I guess the only way to know if replacement is safe would be to get the type from the function parameter, which may be too complex for Biome.

**Comment by arendjr** [comment]
At least for now it is, yes :) We do intend to implement type inference as part of our roadmap, but we haven't started that yet.
